src/interpreter.rs

```diff
@@ -4,12 +4,12 @@ use std::collections::HashMap;
 type Stack<T> = Vec<T>;
 
 #[derive(Default)]
-pub struct Interpreter<'a, T> {
+pub struct Interpreter<'a, 'b, T: 'b> {
```

Comment:
Also, what does `T: 'b` mean? Why is it necessary? `T` is a type, its life-time is compile-time :confused: